File: src/types/project.ts
/**
 * @deprecated Static phase enum - prefer dynamic ProjectPhaseData from API.
 * Java backend stores phases in prj_phase table, not static enum.
 */
export enum ProjectPhase {
  SURVEY = 1,
  CONCEPT = 2,
  PLANNING = 3,
  DEVELOPMENT = 4,
  TESTING = 5,
  MASS_PRODUCTION = 6,
}

/** Dynamic project phase data from API (prj_phase table) */
export interface ProjectPhaseData {
  id: number;
  projectId: number;
  name: string;
  code: string;
  seqNo: number;
  status: number;
  progress: number;
  plannedStart?: string;
  plannedEnd?: string;
  actualStart?: string;
  actualEnd?: string;
}

export enum ProjectStatus {
  CANCELLED = 0,
  NOT_STARTED = 1,
  IN_PROGRESS = 2,
  COMPLETED = 3,
  DELAYED = 4,
}

export interface Project {
  id: number;
  code?: string;
  name: string;
  description?: string;
  parentId?: number | null;
  projectType?: string;
  priority?: number;
  status: ProjectStatus;
  ownerId: number;
  ownerName?: string;
  plannedStart: string;
  plannedEnd: string;
  actualStart?: string;
  actualEnd?: string;
  progress: number;
  parentPath?: string;
  flowInstanceId?: string;
  children?: Project[];
  createdAt: string;
  updatedAt: string;
}

export interface ProjectTreeNode extends Project {
  children?: ProjectTreeNode[];
  level: number;
}

File: syncflow-java/syncflow-project/src/main/java/com/syncflow/project/entity/Project.java
package com.syncflow.project.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * Core project entity for industrial project management.
 * Maps to table {@code prj_project}.
 */
@Data
@TableName("prj_project")
public class Project implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    /** Primary key, auto-increment (BIGSERIAL). */
    @TableId(value = "id", type = IdType.AUTO)
    private Long id;

    /** Project display name. */
    private String name;

    /** Unique project code for programmatic reference. */
    private String code;

    /** Rich-text project description. */
    private String description;

    /** FK to sys_user.id, project owner / manager. */
    private Long ownerId;

    /** Project classification: R&D, PRODUCTION, MAINTENANCE, etc. */
    private String projectType;

    /**
     * Project status.
     * <ul>
     *   <li>1 = not_started</li>
     *   <li>2 = in_progress</li>
     *   <li>3 = completed</li>
     *   <li>4 = delayed</li>
     *   <li>0 = cancelled</li>
     * </ul>
     */
    private Integer status;

    /**
     * Priority level.
     * <ul>
     *   <li>1 = urgent</li>
     *   <li>2 = high</li>
     *   <li>3 = medium</li>
     *   <li>4 = low</li>
     * </ul>
     */
    private Integer priority;

    /** Completion percentage 0-100. */
    private Integer progress;

    /** Planned project start date. */
    private LocalDate plannedStart;

    /** Planned project end date. */
    private LocalDate plannedEnd;

    /** Actual project start date. */
    private LocalDate actualStart;

    /** Actual project end date. */
    private LocalDate actualEnd;

    /** FK to prj_project.id for sub-projects. */
    private Long parentId;

    /** Materialised ancestor path for fast tree queries. */
    private String parentPath;

    /** FK to sys_department.id, owning department. */
    private Long deptId;

    /** Workflow engine instance identifier. */
    private String flowInstanceId;

    /** Tenant identifier for multi-tenancy. */
    private Long tenantId;

    /** Row creation timestamp (auto-filled on insert). */
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    /** Row last-update timestamp (auto-filled on insert and update). */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    /** Soft-delete timestamp, NULL means not deleted. */
    @TableLogic
    private LocalDateTime deletedAt;
}
